### PR TITLE
Add Video Restore feature to docs

### PR DESCRIPTION
**Pull request by szekelyzol**
Changes are for [this Asana task](https://app.asana.com/0/1205634133195403/1208125970959632).

**Summary**

* added information about the feature to the [Delete a video](https://docs.api.video/vod/delete-a-video) and [Video best practices](https://docs.api.video/vod/video-best-practices) guides.
* added some missing internal links to the API reference in the [Delete a video](https://docs.api.video/vod/delete-a-video) guide
* updated the API reference 👇 

**OpenAPI changes**

* added the new query parameter `removed` to `GET /videos` 
* added the new body parameter `removed` to `PATCH /videos/{videoId}`
* added new fields `deletesAt` and `removedAt` to the `video` response object
* updated the description of all date-time fields in the `video` response object to show ATOM format
* removed unnecessary mention of dynamic metadata from the description of the `video` response object

### CODE REVIEW

**Comment by szekelyzol** [comment]
⚠️ @olivierapivideo some URLs in these changes might be incorrect and are pending on [these](https://api-video.slack.com/archives/C07CTDAVD2M/p1724749307134079) [questions](https://api-video.slack.com/archives/C07CTDAVD2M/p1724749368838559) in Slack. 🙏

**Comment by szekelyzol** [comment]
New preview build [here](https://api-video.doctave.dev/_preview/7509/).

File: openapi.yaml [hunk 1]
```diff
@@ -12257,11 +13280,11 @@ paths:
...
                         to: '2024-05-29T11:08:39+00:00'
                     data:
-                    - emittedAt: '2024-05-29T07:00:00+00:00'
+                    - emittedAt: '2024-05-29T07+00:00:00:00'
                       metricValue: 2
-                    - emittedAt: '2024-05-29T08:00:00+00:00'
+                    - emittedAt: '2024-05-29T08+00:00:00:00'
                       metricValue: 1
-                    - emittedAt: '2024-05-29T09:00:00+00:00'
+                    - emittedAt: '2024-05-29T09+00:00:00:00'
```
Comment (olivierapivideo) [inline openapi.yaml:13287]: Not sure this is valid ?

Comment (szekelyzol) [reply openapi.yaml:13287]: Hmm I think you are right, this one is not correct  😬 I'll fix this.

**Review by olivierapivideo — APPROVED** [review]

**Pull request merged**